packages/cli/src/workflows/workflow.service.ts

```diff
@@ -398,6 +406,65 @@ export class WorkflowService {
 		return workflow;
 	}
 
+	async archive(user: User, workflowId: string): Promise<WorkflowEntity | undefined> {
+		const workflow = await this.workflowFinderService.findWorkflowForUser(workflowId, user, [
+			'workflow:delete',
```

Comment:
Why not have a dedicated `workflow:archive` permission?

Reply:
We can do this, but do we want to make archive "soft delete" permission different from deletion permissions? We didn't consider this while planning this feature.

Reply:
Not much context on this feature so up to you. Looked worth bringing up.